[motion-path] circle() and ellipse() should use non-offset starting position

nmoucht opened the pull request:
#### eddbc5df03c84c7da6b202923cad6ae71ecc1b0e
<pre>
[motion-path] circle() and ellipse() should use non-offset starting position
<a href="https://bugs.webkit.org/show_bug.cgi?id=262014">https://bugs.webkit.org/show_bug.cgi?id=262014</a>
rdar://115962433

Reviewed by Tim Nguyen.

circle() and ellipse() should use non-offset starting position. We don&apos;t want to use the offset starting
position as this would make the calculation of the radius size incorrect for circle/ellipse. The one wpt
for circle()/ellipse() do not actually test this properly (they use a 0,0 starting position).

* LayoutTests/imported/w3c/web-platform-tests/css/motion/offset-path-shape-circle-008-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/motion/offset-path-shape-circle-008.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/motion/offset-path-shape-ellipse-007-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/motion/offset-path-shape-ellipse-007.html: Added.
* Source/WebCore/rendering/MotionPath.cpp:
(WebCore::MotionPath::computePathForShape):

Canonical link: <a href="https://commits.webkit.org/268400@main">https://commits.webkit.org/268400@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0fe27222a21d18a0dad96efed78c49dc97604eb5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/19548 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/19968 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/20577 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21440 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18286 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/19785 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23231 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20111 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/19886 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/19765 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/19787 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/16994 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22297 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/16974 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/17782 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/24095 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18035 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/17957 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22067 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/18566 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/15733 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/17709 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/17617 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22066 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2393 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18393 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
Seems fine. Can you add the context you wrote in your comment in the commit message?
> We don't want to use the offset starting position as this would make the calculation of the radius size incorrect for circle/ellipse. The one WPT for circle()/ellipse() do not actually test this properly (they use a 0,0 starting position).

A spec reference would be good if you can find one too.

Regarding cross browser results:

They pass fine in Firefox: https://wpt.fyi/results/css/motion/?diff=&filter=ADC&run_id=5197978878345216&run_id=5112401990516736

These tests appear to fail in Chrome, mostly due to a scrollbar difference: https://wpt.fyi/results/css/motion?diff&filter=ADC&run_id=5086110482432000&run_id=5161220870504448

I think Chrome is incorrect here, I don't see any obvious overflow that would cause a scrollbar to show.

nmoucht committed:
[motion-path] circle() and ellipse() should use non-offset starting position
https://bugs.webkit.org/show_bug.cgi?id=262014
rdar://115962433

Reviewed by Tim Nguyen.

circle() and ellipse() should use non-offset starting position. We don't want to use the offset starting
position as this would make the calculation of the radius size incorrect for circle/ellipse. The one wpt
for circle()/ellipse() do not actually test this properly (they use a 0,0 starting position).

* LayoutTests/imported/w3c/web-platform-tests/css/motion/offset-path-shape-circle-008-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/motion/offset-path-shape-circle-008.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/motion/offset-path-shape-ellipse-007-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/motion/offset-path-shape-ellipse-007.html: Added.
* Source/WebCore/rendering/MotionPath.cpp:
(WebCore::MotionPath::computePathForShape):

Canonical link: https://commits.webkit.org/268400@main
